nncf/torch/model_graph_manager.py

```diff
@@ -337,3 +337,28 @@ def get_fake_quantizer(
             if isinstance(module, BaseQuantizer):
                 return module
     return None
+
+
+def get_target_dim_for_weight_compression(metatype: om.PTOperatorMetatype, ndims: int, input_port_id: int) -> int:
+    """
+    Determines the target dimension for weight compression.
+
+    :param metatype: The node metatype for which the target dimension is being determined.
+    :param input_port_id: The input port id.
+    :return: The target dimension for weight compression.
+    """
+    if metatype == om.PTAddmmMetatype:
+        if input_port_id == 1:
+            return 0 if ndims == 1 else ndims - 2
+        if input_port_id == 2:
+            return ndims - 1
+        raise ValueError(f"Unexpected {input_port_id=} for {metatype=}")
+    if metatype == om.PTMatMulMetatype:
```

Comment:
Please check that you support all cases from https://pytorch.org/docs/stable/generated/torch.matmul.html